pkg/controllers/awsloadbalancercontroller/rbac_clusterrole.go

```diff
@@ -0,0 +1,125 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+// ensureClusterRole creates and maintains a clusterrole required by the aws-load-balancer controller.
+func (r *AWSLoadBalancerControllerReconciler) ensureClusterRole(ctx context.Context, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredClusterRole(ctx)
+
+	reqLogger.Info("ensuring clusterroles", "clusterroles", desired.Name)
+
+	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for clusterroles %s : %w", desired.Name, err)
+	}
+
+	exist, current, err := r.currentClusterRole(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current clusterroles: %w", err)
+	}
+
+	if !exist {
+		if err := r.createClusterRole(ctx, desired); err != nil {
+			return err
+		}
+		_, current, err = r.currentClusterRole(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created clusterroles: %w", err)
+		}
+		reqLogger.Info("created clusterroles", "clusterroles", desired.Name)
+	}
+
+	if err := r.updateClusterRole(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update clusterroles: %w", err)
+	}
+
+	return nil
+}
+
+// updateClusterRoles updates the current clusterroles and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateClusterRole(ctx context.Context, current *rbacv1.ClusterRole, desired *rbacv1.ClusterRole) error {
+	changed, err := isClusterRoleChanged(current, desired)
+	if err != nil {
+		return err
+	}
+
+	if !changed {
+		return nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
```

Comment:
Setting name and namespace is not required because it's not possible. Also update with a new name and namespace will fail.

Reply:
Addressed.